x/staking/src/types/domain.rs

```diff
@@ -256,6 +265,56 @@ impl Validator {
             .checked_div(Decimal256::new(self.tokens))?)
     }
 
+    pub fn shares_from_tokens_truncated(&self, amount: Uint256) -> anyhow::Result<Decimal256> {
+        if self.tokens.is_zero() {
+            return Err(AppError::Custom("insufficient shares".into()).into());
+        }
+
+        // TODO: check
+        let mul = self.delegator_shares.checked_mul(Decimal256::new(amount))?;
+        // TODO: check constant 18 in decimals
+        let precision_reuse = Decimal256::new(Uint256::from_u128(10u128)).checked_pow(18)?;
```

Comment:
`Decimal256::new(x)` has slightly unexpected behaviour e.g. `Decimal256::new(1)` "=" 0.000000000000000001. Is this the behaviour you want here?

Reply:
If you want 1 use `Decimal256::from_atomics(1, 0)`

Reply:
I expected another behavior. I'll check the code and fix the places with such constructors.

Reply:
It was unexpected to me, too.